spec/client/ics-008-wasm-client/README.md

```diff
@@ -0,0 +1,447 @@
+---
+ics: 8
+title: Wasm Client
+stage: draft
+category: IBC/TAO
+kind: instantiation
+implements: 2
+author: Parth Desai <parth@chorus.one>, Mateusz Kaczanowski <mateusz@chorus.one>
+created: 2020-10-13
+modified: 2020-10-13
+---
+
+## Synopsis
+
+This specification document describes an interface to a client (verification algorithm) stored as a Wasm bytecode for a blockchain.
+
+### Motivation
+
+Wasm based light clients are decoupled from SDK source code, which enables one to upgrade existing light client or add support for other blockchain without modifying SDK source code.
+
+### Definitions
+
+Functions & terms are as defined in [ICS 2](../ics-002-client-semantics).
+
+`currentTimestamp` is as defined in [ICS 24](../ics-024-host-requirements).
+
+`Wasm Client Code` refers to Wasm bytecode stored in the client store, which provides a target blockchain specific implementation of [ICS 2](../ics-002-client-semantics).
+
+`Wasm Client` refers to a particular instance of `Wasm Client Code` defined as a tuple `(Wasm Client Code, ClientID)`.
+
+`Wasm VM` refers to a virtual machine capable of executing valid Wasm bytecode.
+
+
+### Desired Properties
+
+This specification must satisfy the client interface defined in ICS 2.
+
+## Technical Specification
+
+This specification depends on the correct instantiation of the `Wasm client` and is decoupled from any specific implementation of the target `blockchain` consensus algorithm.
+
+### Client state
+
+The Wasm client state tracks location of the Wasm bytecode via `codeId`. Binary data represented by `data` field is opaque and only interpreted by the Wasm Client Code. `type` represents client type.
+`type` and `codeId` both are immutable.
+
+```typescript
+interface ClientState {
+  codeId: []byte
+  data: []byte
+  latestHeight: Height
+  proofSpecs: []ProofSpec
+}
+```
+
+### Consensus state
+
+The Wasm consensus state tracks the timestamp (block time), `Wasm Client code` specific fields and commitment root for all previously verified consensus states.
+`type` and `codeId` both are immutable.
+
+```typescript
+interface ConsensusState {
+  codeId: []byte
+  data: []byte
+  timestamp: uint64
+}
+```
+
+### Height
+
+The height of a Wasm light client instance consists of two `uint64`s: the epoch number, and the height in the epoch.
+
+```typescript
+interface Height {
+  epochNumber: uint64
```

Comment:
This should be revision number and revision height?

Reply:
yes, will update